### ISSUE TITLE
Search for a Property/Event/Method that has been explicitly set

### ISSUE THREAD

**Issue by myearwood1** [VFPX/GoFish#203]
## 📝 Search for Properties/Events/Methods that have been explicitly set

In a framework-based project, a single PEM can be set across many forms. If one or more programmers made mistakes in one or more of a specific PEM, it is valuable to be able to see all occurrences.

This feature was known as XXDTExplicitPEM. 

---

If you'd like to see this feature implemented, add a 👍 reaction to this post.

**Comment by Jimrnelson** [comment]
@myearwood1 

Could you elaborate of your request?  I do not understand what you're asking for, so I have questions:

- Are you saying that GF is not currently finding these assignments?
- Is there something special about the framework that you expect GF to know about?
- What do you mean by "explicitly set"?
- What is "XXDTExplicitPEM"?

Perhaps an example or two would be enlightening.

**Comment by myearwood1** [comment]
xxdtexplicitpem was the name of the .prg that provided the feature. Explicitly set - A property/event/method that has been specifically overridden, not left at default value. GF finds all search results, it cannot deliver a list of a all modules wherein a named property/event/method is overridden.

**Comment by Jimrnelson** [comment]
When you say  _A property/event/method that has been specifically overridden_, do you mean in the properties window?

If not, where?

**Comment by myearwood1** [comment]
Yes, in the same way that the property sheet shows overridden PEMS vs all, except to show that a named PEM has been overridden. The code goes through the vcx/scx and picks anything where the named PEM is not empty.

**Comment by Jimrnelson** [comment]
Let's address properties first.

If you use GF to search for a property name, it will find all references to that property in code as well as all files (SCX/VCX) where it shows as non-default in the property sheet.

Matches to non-default values show in the grid with the value '\<Property Name\>' in the MatchType column.

You can use the filter checkbox:

![image](https://github.com/VFPX/GoFish/assets/81179335/ef6b0c21-820c-4f14-9009-2acb89a9f27b)

to find all records with that value in the MatchType column:

![image](https://github.com/VFPX/GoFish/assets/81179335/0d42ed65-e180-4108-acfc-cf2ae1c2f86e)

Note that doing so will also filter the TreeView on the left.

It seems to me that this is exactly what you are looking for.  If not, please be very specific about what this does not provide.

**Comment by myearwood1** [comment]
I have never been able to use Match Types filter. Considering how I am treated like a pariah, I don't bother expecting any help from any of you. I have never been as sickeningly disgusting and in public as one of the so-called FoxPro leaders. I offer help and am met with insults and libel.

**Comment by Jimrnelson** [comment]
Mike --

Please explain what does not work for you when using the Match Types filter.  If there is a problem with it that you are experiencing, I want to fix it.

It might be helpful to include some screenshots so that I can tell what you're doing.

**Comment by myearwood1** [comment]
![image](https://github.com/VFPX/GoFish/assets/5750619/52a1518d-8f16-4604-976a-9a27a003ead6)

OK, two birds with one stone then. Why don't you instantiate GF_SearchResult only once when the form loads? Createobject is one of the slowest things. 

So I search for CreateObject and click filter and what I see is not very useful. I would have done it by including the filtering in the initial pass through all the files rather than filtering the returned set.

**Comment by Jimrnelson** [comment]
I thought you were searching for non-default properties.  

What do you get when you search for one of them that are of interest to you?

**Comment by myearwood1** [comment]
I wish I could filter at all. Here's a screenshot from a search for the SetBo method of a MaxFrame application. Filter on Object name for "Address" would be perfect.
![image](https://github.com/VFPX/GoFish/assets/5750619/c2fd22d6-a7f8-4bec-b3c6-df9614ad6942)

**Comment by Jimrnelson** [comment]
Apparently you already have filters on both the Code and Names pages.  (Filters are persistent).

I suggest you start with "Clear All" and then set the filters you want.

If you want to filter just on objects named "Address", you should be able to do that on the "Names" page.

And I wouldn't think the Match Type would be relevant if you were looking for objects names "Address".

**Comment by myearwood1** [comment]
I figured out how to do what I wanted. I searched for setbo. I then added a filter by using the Names tab and put *address* and checked use wildcards. This ticket can be closed. Please figure out how to add it to documentation.

**Comment by Jimrnelson** [comment]
I am marking this as closed and opened a new issue in response.
